third_party/badvpn/socks_udp_client/SocksUdpClient.c

```diff
@@ -424,6 +452,14 @@ static int compute_mtu(int udp_mtu)
     return udp_mtu + sizeof(struct socks_udp_header) + sizeof(struct socks_addr_ipv6);
 }
 
+static int dns_id(BAddr *remote_addr, const uint8_t *data, int data_len)
```

Comment:
nit: rename to use a verb, i.e. `get_dns_id`

Reply:
Could you leverage this implementation to extract the query ID? https://github.com/Jigsaw-Code/outline-client/blob/master/third_party/badvpn/misc/dns_proto.h#L54

Reply:
Done